examples/typescript-node-es6/src/index.ts

```diff
@@ -75,6 +75,14 @@ new ParseError('message', {}, domException);
 assert(Node.ATTRIBUTE_NODE, 2);
 assert(Node.DOCUMENT_POSITION_CONTAINS, 8);
 
+// there are no real Node instances,
+// but we want to check that the Node type provides these props
+const fakeNode = {} as unknown as Node;
+assert(fakeNode.nodeType, undefined);
+assert(fakeNode.lineNumber, undefined);
+assert(fakeNode.columnNumber, undefined);
+assert(fakeNode.textContent, undefined);
```

Comment:
As I mentioned in the comment above this section, yes, there is no actual instance of type `Node`, but these assertions should make sure that those properties are available on the type `Node`, not only on some sub type/class.
When using `assert`, the CodeQL checks will complain that the values are unused.
And yes, I'm asserting that the value is undefined, which only makes sense in the context of this assertion.

Reply:
Ahh I think I get it now... so basically if you would try to do this: `assert(fakeNode.doesntExist, undefined)` JS would be fine with it but since the TS compiler checks this the check would fail correctly.